src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -96,6 +96,12 @@ trait GenTraversableOnce[+A] extends Any {
    */
   def size: Int
 
+   /** The size of this $coll if it is can be cheaply computed
+   *
+   *  @return    the number of elements in this $coll, or -1 if the size cannot be determined cheaply
+   */
+  protected[collection] def sizeHint: Int = -1
```

Comment:
`sizeInfo`? Assuming we don't want to introduce some kind of more sophisticated design as in psp?